config/routes.rb

```diff
@@ -10,7 +10,11 @@
   resources :movies
   resources :halls
   resources :reservations, excep: %i[show new create]
+
   post 'screenings/:screening_id/reservations/at_desk', to: 'reservations#create_at_desk', as: 'reservation_at_desk'
+  post 'screenings/:screening_id/reservations/create_without_registration',
+       to: 'reservations#create_without_registration', as: 'reservation_without_registration'
+
```

Comment:
these routes are a mess 😅 it would be better to include them in a `resources` so we will know better what routes we have for specific model:

```ruby
  resources :screenings do
    resources :reservations, only: %i[new create] do
      collection do
        post 'create_at_desk', to: 'reservations#create_at_desk'
        post 'create_without_registration', to: 'reservations#create_without_registration'
      end
    end
  end
  resources :movies
  resources :halls
  resources :reservations, except: %i[show new create] do
    patch 'cancel', to: 'reservations#cancel'
    patch 'confirm', to: 'reservations#confirm'
  end
```

and you don't need `as` argument because rails will create these paths and usually we don't create this especially in API only apps because these are helpers for views and redirections and we don't use them in API only apps :)

Reply:
Thanks a lot for looking it up, I was also thinking that it should be a way to make it more clean but wasn't too sure how :)